identity-wallet/src/state/search/reducers/add_recent_search.rs

```diff
@@ -0,0 +1,31 @@
+use crate::error::AppError;
+use crate::state::search::actions::add_recent_search::AddRecentSearch;
+use crate::state::{
+    actions::{listen, Action},
+    AppState,
+};
+
+const MAX_RECENT_SEARCHES: usize = 3;
+
+pub async fn add_recent_search(state: AppState, action: Action) -> Result<AppState, AppError> {
+    if let Some(recent_search) = listen::<AddRecentSearch>(action) {
+        let mut search_results = state.search_results;
+
+        search_results
+            .recent_credentials
+            .retain(|recent| recent != &recent_search.id);
+        search_results.recent_credentials.insert(0, recent_search.id);
+
+        if search_results.recent_credentials.len() > MAX_RECENT_SEARCHES {
+            search_results.recent_credentials.remove(MAX_RECENT_SEARCHES);
```

Comment:
This code is fine, but .pop(); will always ensure the last one is removed

Reply:
good comment! just merged it already, sorry

Reply:
haha ok no problem last thing I saw:

```
  // makes the <input> element available to the parent component (example usage: conditionally put focus it)
```
On a svelte export let. we can change comment style to /** mak.... */ to work with intellisense but for now its ok :+1: